plugins/modules/homebrew_services.py

```diff
@@ -0,0 +1,260 @@
+description:
+    - Manages daemons and services via Homebrew
+extends_documentation_fragment:
+    - community.general.attributes
+options:
+    name:
+        description:
+            - An installed homebrew package whose service is to be updated.
+        aliases: [ 'formula', 'package', 'pkg' ]
+        type: str
+    path:
+        description:
+            - "A V(:) separated list of paths to search for C(brew) executable.
+              Since a package (I(formula) in homebrew parlance) location is prefixed relative to the actual path of C(brew) command,
+              providing an alternative C(brew) path enables managing different set of packages in an alternative location in the system."
+        default: '/usr/local/bin:/opt/homebrew/bin:/home/linuxbrew/.linuxbrew/bin'
+        type: path
+    state:
+        description:
+            - state of the package.
+        choices: [ 'present', 'absent', 'restarted' ]
+        default: present
+        type: str
+"""
+
+EXAMPLES = """
+# Install foo package
+- community.general.homebrew:
+    name: foo
+    state: present
+
+# Start the foo service (equivalent to C(brew services start foo))
+- community.general.homebrew_service:
+    name: foo
+    state: present
+
+# Restart the foo service (equivalent to C(brew services restart foo))
+- community.general.homebrew_service:
+    name: foo
+    state: restarted
+
+# Remove the foo service (equivalent to C(brew services stop foo))
+- community.general.homebrew_service:
+    name: foo
+    service_state: absent
+"""
+
+RETURN = """
+msg:
+    description:
+      - Whether the service is now running, its PID if it is running,
+        and whether the running state changed (including restarts).
+    returned: always
+    type: str
+    sample: "Running: true, Changed: false, PID: 1234"
+pid:
+    description:
+      - If the service is now running, this is the PID of the service, otherwise -1.
+    returned: success
+    type: int
+    sample: 1234
+running:
+    description:
+      - Whether the service is running after running this command.
+    returned: success
+    type: bool
+    sample: true
+changed:
+    description:
+      - Whether the service state changed. This is true when an already running service is restarted.
+    returned: success
+    type: bool
+    sample: false
+"""
+
+from dataclasses import dataclass
+from typing import Callable, Optional
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.homebrew import (
+    HomebrewValidate, parse_brew_path
+)
+
+
+@dataclass
+class HomebrewServiceArgs:
+    """Stores validated arguments for an instance of an action.
+
+    See DOCUMENTATION string for argument-specific information.
+    """
+
+    name: str
+    state: str
+    brew_path: str
+
+
+@dataclass
+class HomebrewServiceState:
+    """Stores the state of a Homebrew service."""
+
+    running: bool
+    pid: Optional[int]
```

Comment:
Please note that modules still have to support Python 2.7, which neither has typing nor dataclasses. (Python 3.5 and 3.6 also don't have dataclasses BTW, and are also still supported.)

A Python 2.7 compatible alternative are named tuples: https://docs.python.org/2/library/collections.html#collections.namedtuple

Reply:
Ah can do. Apologies! I think I misunderstood the documentation [here](https://docs.ansible.com/ansible/latest/community/collection_contributors/collection_requirements.html#python-requirements) and [here](https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html#ansible-core-support-matrix). I had assumed that the modules don't have to run on the target machine, and so only I had only considered the Control Node column.